src/components/modals/ContactsPermissionModal.tsx

```diff
@@ -16,15 +22,52 @@ interface Props {
   bridge: AirshipBridge<ContactsPermissionResult | undefined>
 }
 
+export let showModalPromise: Promise<ContactsPermissionResult | undefined>
+
+/**
+ * Shows one instance of this modal asking if the user wants to grant contacts
+ * permissions, if the system contacts permission isn't granted.
+ *
+ * @returns Result of Contacts Access modal or undefined if no modal shown or
+ * the modal was dismissed somehow.
+ */
+export async function maybeShowContactsPermissionModal(): Promise<ContactsPermissionResult | undefined> {
+  const contactsPermissionOn = (await check(permissionNames.contacts).catch(_error => 'denied')) === 'granted'
+  if (contactsPermissionOn || showModalPromise != null) return undefined
+
+  showModalPromise = (async () => {
+    return await Airship.show<ContactsPermissionResult | undefined>(bridge => <ContactsPermissionModal bridge={bridge} />)
+  })()
+
+  return await showModalPromise
+}
+
+/**
+ * Shows the modal if it hasn't been shown before, and attempts to set the
+ * system contacts permission setting
+ */
```

Comment:
We don't need to `export ContactsPermissionModal` anymore, since the only way to show it is via `maybeShowContactsPermissionModal`.

In the future, if there are scenarios where we want to show the modal a second or third time, we add a parameter like `maybeShowContactsPermissionModal(showAgain? boolean)`, since even in these cases, we would still want all the other logic that comes with the function.